# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/test/gtest/parent/datamodel/TestFileSystemDataManager.cpp
# Commit: 6d1f25188b13
# Full Hash: 6d1f25188b13f5c88f36bcb32caa2b74216f81ab
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-08-31 21:54:47
# Regressor Bug: 1786465
# File Overlap Count: 2
# Description:
#   Bug 1786465 - Add support for asynchronous closing of FileSystemDataManager; r=dom-storage-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155787
# ==============================================================================

diff -r 0256861a7ac5 -r 6d1f25188b13 dom/fs/test/gtest/parent/datamodel/TestFileSystemDataManager.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/fs/test/gtest/parent/datamodel/TestFileSystemDataManager.cpp	Wed Aug 31 09:34:58 2022 +0000
@@ -0,0 +1,39 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "FileSystemDataManager.h"
+#include "gtest/gtest.h"
+#include "mozilla/SpinEventLoopUntil.h"
+#include "TestHelpers.h"
+
+namespace mozilla::dom::fs::test {
+
+namespace {
+
+constexpr auto kTestOriginName = "http:://example.com"_ns;
+
+}  // namespace
+
+TEST(TestFileSystemDataManager, GetOrCreateFileSystemDataManager)
+{
+  TEST_TRY_UNWRAP(Registered<data::FileSystemDataManager> registeredDataManager,
+                  data::FileSystemDataManager::GetOrCreateFileSystemDataManager(
+                      Origin(kTestOriginName)));
+
+  RefPtr<data::FileSystemDataManager> dataManager = registeredDataManager.get();
+
+  registeredDataManager = nullptr;
+
+  bool done = false;
+
+  dataManager->OnClose()->Then(
+      GetCurrentSerialEventTarget(), __func__,
+      [&done](const BoolPromise::ResolveOrRejectValue&) { done = true; });
+
+  SpinEventLoopUntil("Promise is fulfilled"_ns, [&done]() { return done; });
+}
+
+}  // namespace mozilla::dom::fs::test